test/e2e/network/endpointslice.go

```diff
@@ -492,8 +497,12 @@ func hasMatchingEndpointSlices(cs clientset.Interface, ns, svcName string, numEn
 		framework.Logf("EndpointSlice for Service %s/%s not found", ns, svcName)
 		return []discoveryv1beta1.EndpointSlice{}, false
 	}
-	if len(esList.Items) != numSlices {
-		framework.Logf("Expected %d EndpointSlices for Service %s/%s, got %d", numSlices, ns, svcName, len(esList.Items))
+	// In some cases the EndpointSlice controller will create more
+	// EndpointSlices than necessary resulting in some duplication. This is
+	// valid and tests should only fail here if less EndpointSlices than
+	// expected are added.
+	if len(esList.Items) < numSlices {
```

Comment:
do you have to modify any of the callers of `hasMatchingEndpointSlices` with the `numSlices`?

Reply:
I don't think so. I changed this function from "has exactly n slices" to "has at least n slices", so all previously passing tests should still pass here.